### PR TITLE
Feature/phpstan

### PR DESCRIPTION
**Pull request by jeroenjans**
This PR adds support for [PHPStan](https://github.com/phpstan/phpstan) using the CheckStyleParser.

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/analysis/warnings/PhpStan.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+/**
+ * Provides a parser and customized messages for PHPStan. Delegates to {@link CheckStyleParser}.
+ *
+ * @author Jeroen Jans
+ */
+public class PhpStan extends ReportScanningTool {
+    private static final long serialVersionUID = 2699509705079011738L;
+
+    static final String ID = "phpstan";
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/analysis/warnings/PhpStan.java]: private

Comment (uhafner) [reply src/main/java/io/jenkins/plugins/analysis/warnings/PhpStan.java]: (I changed that in all other parsers as well:-)

Comment (jeroenjans) [reply src/main/java/io/jenkins/plugins/analysis/warnings/PhpStan.java]: I've updated the visibility to private.
I've also rebased on master to resolve the merge conflict in the SUPPORTED-FORMATS.md file

**Comment by jeroenjans** [comment]
Regarding the image I haven't been able to find information on the license of the icon.
I'll see if I can get some information from the developers on this.

**Comment by jeroenjans** [comment]
Alright, I was able to check it with the developers and I was told that we can use the logo for this purpose.
I've added it in the last commit and updated the supported documents file.

**Comment by uhafner** [comment]
Nice, thanks! Can you please add a line or reference to the icon license (or to the discussion) in the LICENSE.txt file in the icons folder?

**Comment by uhafner** [comment]
And one more thing, can you add a small test case to [ParsersITest](https://github.com/jenkinsci/warnings-ng-plugin/blob/master/src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java#L70) that shows that your parser correctly finds *n* warnings. No need to check for individual properties, since this is done in the check style parser test.

**Comment by uhafner** [comment]
Thank you very much!

**Pull request merged**